src/Compilers/CSharp/Portable/Compiler/AnonymousTypeMethodBodySynthesizer.cs

```diff
@@ -191,30 +189,16 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                 //  prepare symbols
                 MethodSymbol equalityComparer_GetHashCode = manager.System_Collections_Generic_EqualityComparer_T__GetHashCode;
                 MethodSymbol equalityComparer_get_Default = manager.System_Collections_Generic_EqualityComparer_T__get_Default;
-                NamedTypeSymbol equalityComparerType = equalityComparer_GetHashCode.ContainingType;
 
                 //  bound HASH_FACTOR
-                BoundLiteral boundHashFactor = F.Literal(HASH_FACTOR);
+                BoundLiteral boundHashFactor = null;
 
                 // Process fields
                 for (int index = 0; index < anonymousType.Properties.Length; index++)
```

Comment:
Please consider using `foreach`.

Reply:
> Please consider using foreach.

This is existing code.

---
In reply to: [437728180](https://github.com/dotnet/roslyn/pull/44992#discussion_r437728180) [](ancestors = 437728180)